src/SAML2/AuthnRequestFactory.php

```diff
@@ -39,7 +40,27 @@ class AuthnRequestFactory
     public static function createFromHttpRequest(Request $httpRequest)
     {
         // the GET parameter is already urldecoded by Symfony, so we should not do it again.
-        $samlRequest = gzinflate(base64_decode($httpRequest->get(AuthnRequest::PARAMETER_REQUEST)));
+        $samlRequest = base64_decode($httpRequest->get(AuthnRequest::PARAMETER_REQUEST), true);
+        if ($samlRequest === false) {
+            throw new InvalidRequestException('Failed decoding the request, did not receive a valid base64 string');
+        }
+
+        // gzinflate throws an error
+        $errorNo = $errorMessage = null;
```

Comment:
You ill, mate? This is very unlike yourself :stuck_out_tongue: I have no problems with it, though :wink:

The comment can go or maybe be rephrased to "Catch any errors gzinflate triggers" or "gzinflate may trigger an error".